src/mysql_vm_helpers.py

```diff
@@ -163,6 +188,41 @@ def wait_until_mysql_connection(self) -> None:
         if not os.path.exists(MYSQLD_SOCK_FILE):
             raise MySQLServiceNotRunningError()
 
+    def connect_mysql_exporter(self) -> None:
+        """Set up mysqld-exporter config options.
+
+        Raises
+            snap.SnapError: if an issue occurs during config setting or restart
+        """
+        cache = snap.SnapCache()
+        mysqld_exporter = cache[MYSQL_EXPORTER_SNAP_NAME]
+
+        try:
+            # Set up exporter with connection info and disable metrics
+            mysqld_exporter.set(
+                {
+                    "mysql.socket": MYSQLD_SOCK_FILE,
+                    "mysql.user": self.monitoring_user,
+                    "mysql.password": self.monitoring_password,
+                    "collect.auto-increment.columns": "false",
+                    "collect.info-schema.tables": "false",
+                    "collect.info-schema.tablestats": "false",
+                    "collect.perf-schema.indexiowaits": "false",
+                    "collect.perf-schema.tableiowaits": "false",
+                    "collect.perf-schema.tablelocks": "false",
+                    "collect.info-schema.userstats": "false",
+                    "collect.binlog-size": "false",
+                    "collect.info-schema.processlist": "false",
+                }
+            )
+            mysqld_exporter.restart()
+        except snap.SnapError as e:
+            logger.error(
+                "An exception occurred when setting configs for mysqld-exporter snap. Reason: %s"
+                % e.message
+            )
+            raise
```

Comment:
minor nit: we have been raising custom errors for methods in this file so we can distinguish in the charm file where the exception originated and have conditional logic based on the method's exception

Reply:
8aba997